062-prometheus-metrics-reporter.md

```diff
@@ -56,40 +56,66 @@ metricsConfig:
 ```
 
 Reporters will expose the following configurations:
-- `prometheus.metrics.reporter.port`: The HTTP port to expose the metrics. Default: `8080`
-- `prometheus.metrics.reporter.allowlist`: A comma separated list of regex patterns to specify the metrics to collect. Default: `.*`. The patterns must match the metric names emitted by the reporter.
+- `prometheus.metrics.reporter.port`: The HTTP port to expose the metrics. Default: `8080`. When set to `-1`, the reporter will not start the HTTP endpoint, this can be useful for applications embedding Kafka clients and that have their own mechanism to export metrics.
```

Comment:
Have we considered supporting dynamic port allocation (port 0)?  It might be useful for applications that embed a kafka broker (kafka supports this itself IIRC).   There would need to be a way for the assigned port to be discovered for it to be useful.

Personally, I don't think we actually need this now, so it would be sufficient just to state that it is out of scope.  It could always be a future enhancement should the use-case ever arrive.

Reply:
I see you are using port 0 in tests already.

Reply:
I thought this was implied but we can update the configuration description to explicitly mention it.

Reply:
Done